File: codes/yunyeong/643_maximumaveragesubarray1.py
class Solution:
    def findMaxAverage(self, nums: list[int], k: int) -> float:
        max_avg = min(nums)
        for i in range(0, len(nums) - k + 1 ):
            window = nums[i:i + k]
            avg = sum(window) / k 
            if max_avg < avg:
                max_avg = avg
        return round(max_avg, 5) 
# Time Limit exceeded.

File: codes/yujo/643_maximumaveragesubarray1.ts
// Time Limit Exceeded
function findMaxAverage(nums: number[], k: number): number {
  let max = -Infinity;

  for (let i = 0; i < nums.length - k + 1; i++) {
    let sum = 0;
    for (let j = 0; j < k; j++) {
      sum += nums[i + j];
    }

    max = Math.max(max, sum);
  }

  return max / k;
}